colossalai/fx/passes/algorithms/ckpt_solver_rotor.py

```diff
@@ -5,9 +5,26 @@
 import math
 from .linearize import linearize
 from .operation import ForwardCheck, ForwardEnable, ForwardNograd, Backward, Loss, Chain, Sequence, Function
-from colossalai.fx.passes.meta_info_prop import MetaInfoProp
 from colossalai.fx.codegen.activation_checkpoint_codegen import _find_nested_ckpt_regions
-from colossalai import META_COMPATIBILITY
+try:
+    from .dynamic_programs_C_version import persistent_compute_table
+    CVERSION = True
+except ModuleNotFoundError:
+    import subprocess
+    import os
+    print("dynamic_programs_C_version hasn't been built! Building library...")
+    this_dir = os.path.dirname(os.path.abspath(__file__))
+    result = subprocess.Popen(f'python {os.path.join(this_dir, "build_c_ext.py")} build_ext --build-lib={this_dir}',
+                              stdout=subprocess.PIPE,
+                              stderr=subprocess.PIPE,
+                              shell=True)
+    if result.wait() == 0:
+        print("dynamic_programs_C_version has been built!")
+        from .dynamic_programs_C_version import persistent_compute_table
+        CVERSION = True
+    else:
+        print("dynamic_programs_C_version built failed! Using python version!")
+        CVERSION = False
```

Comment:
This should be put inside the `solver_rotor` function.

Reply:
okay